lib/class-wp-rest-block-types-controller.php

```diff
@@ -0,0 +1,326 @@
+
+		$fields = $this->get_fields_for_response( $request );
+		$data   = array();
+		if ( in_array( 'name', $fields, true ) ) {
+			$data['name'] = $block_type->name;
+		}
+
+		if ( in_array( 'attributes', $fields, true ) ) {
+			$data['attributes'] = $block_type->get_attributes();
+		}
+
+		if ( in_array( 'is_dynamic', $fields, true ) ) {
+			$data['is_dynamic'] = $block_type->is_dynamic();
+		}
+
+		$extra_fields = array( 'editor_script', 'script', 'editor_style', 'style' );
+		foreach ( $extra_fields as $extra_field ) {
+			if ( in_array( $extra_field, $fields, true ) ) {
+				$data[ $extra_field ] = (string) $block_type->$extra_field;
+			}
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+
+		$response->add_links(
+			array(
+				'collection'              => array(
+					'href' => rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ),
+				),
+				'https://api.w.org/items' => array(
+					'href' => rest_url( sprintf( '%s/%s/%s', $this->namespace, $this->rest_base, $block_type->name ) ),
+				),
+			)
+		);
+
+		/**
+		 * Filters a block type returned from the REST API.
+		 *
+		 * Allows modification of the block type data right before it is returned.
+		 *
+		 * @param WP_REST_Response $response   The response object.
+		 * @param object           $block_type The original block type object.
+		 * @param WP_REST_Request  $request    Request used to generate the response.
+		 */
+		return apply_filters( 'rest_prepare_block_type', $response, $block_type, $request );
+	}
+
+	/**
+	 * Retrieves the block type' schema, conforming to JSON Schema.
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'block_type',
+			'type'       => 'object',
+			'properties' => array(
+				'name'          => array(
+					'description' => __( 'The title for the block type.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'attributes'    => array(
+					'description' => __( 'Block attributes.', 'gutenberg' ),
+					'type'        => 'object',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'is_dynamic'    => array(
+					'description' => __( 'Is the block dynamically rendered.', 'gutenberg' ),
+					'type'        => 'boolean',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'editor_script' => array(
+					'description' => __( 'URL of editor script js file.', 'gutenberg' ),
```

Comment:
Nit: "script" and "js" seem redundant with eachother. Personally, I'd also avoid abbreviations.

Maybe:

```suggestion
					'description' => __( 'URL of editor script.', 'gutenberg' ),
```

(Same applies for a few additional fields below)

Reply:
I was reading [this comment in the Trac ticket](https://core.trac.wordpress.org/ticket/47620#comment:15). I agree with the consensus there to use the script handle. But then, I think these labels should be updated to not refer to this value as the "URL" ?

```suggestion
					'description' => __( 'Editor script handle.', 'gutenberg' ),
```